### ISSUE TITLE
su_user broken when user shell is set to nologin

### ISSUE THREAD

**Issue by artizirk** [Fizzadar/pyinfra#437]
**Describe the bug**
After commit 0c7234be0c3d02172da63f843260851b7ff66794 `su_user` has become broken on systems where target user shell is set to [`nologin`](https://manpages.debian.org/buster/login/nologin.8.en.html).

**To Reproduce**
Target is a Debian server that has a user with shell set to nologin (Default for most service users).

```
root@http:~# grep daemon /etc/passwd
daemon:x:1:1:daemon:/usr/sbin:/usr/sbin/nologin
```

```python
# test.py
from pyinfra.operations import server
server.shell(su_user="daemon", commands=["echo Hello World"])
```

```
$ pyinfra --user root -vv http test.py
--> Loading config...
--> Loading inventory...

--> Connecting to hosts...
    [http] Connected

--> Preparing operations...
    Loading: test.py
    [http] Ready: test.py

--> Proposed changes:
    Ungrouped:
    [http]   Operations: 1   Commands: 1

--> Beginning operation run...
--> Starting operation: Server/Shell ("commands=['echo Hello World']",)
[http] >>> su daemon -c 'sh -c '"'"'echo Hello World'"'"''
    [http] This account is currently not available.
    [http] Error
--> pyinfra error: No hosts remaining!
```

**Expected behavior**
Under Linux, [`su`](https://manpages.debian.org/buster/util-linux/su.1.en.html) should be called with `-s /bin/sh` to bypass `nologin` restriction.

FreeBSD [`su`](https://www.freebsd.org/cgi/man.cgi?query=su&apropos=0&sektion=0&manpath=FreeBSD+12.1-RELEASE+and+Ports&arch=default&format=html) probably needs to use `-m` option but I cant test that.

**Meta**
pyinfra is installed via pip in a virtualenv

```
$ pyinfra --support
--> Support information:

    If you are having issues with pyinfra or wish to make feature requests, please
    check out the GitHub issues at https://github.com/Fizzadar/pyinfra/issues .
    When adding an issue, be sure to include the following:

    System: Linux
      Platform: Linux-4.19.0-9-amd64-x86_64-with-debian-10.5
      Release: 4.19.0-9-amd64
      Machine: x86_64
    pyinfra: v1.1.2
    Executable: /home/arti/infra/venv/bin/pyinfra
    Python: 3.7.3 (CPython, GCC 8.3.0)
```

**Comment by artizirk** [comment]
Downgrading to v1.1.dev1 fixes this
```
pip install pyinfra==1.1.dev1
```

**Comment by Fizzadar** [comment]
This is an interesting one! For an immediate solution, it should be possible to achieve the same using `sudo_user` which should work irrespective of the target user shell.

Back on `su_user` - I see a three possible solutions:

- Detect Linux vs. BSD and use appropriate flags; the `-m` flag leaves the environment similar to `sudo -E` (`preserve_sudo_env`)
- Add an explicit `preserve_su_env` argument that passes `-m` - this is compatible on both Linux/BSD
- Add/or an explicit `su_shell` argument that passes `-l` and would be expected to fail on BSD

The first has a couple of issues - differing behaviour on BSD vs. Linux and BSD would, by default, preserve the environment which may not be desired.

Frustratingly I cannot find any way to set just the shell for BSD `su` variants :/

**Comment by Fizzadar** [comment]
Going to go for adding both explicit arguments here, such that the user has control over this.

**Comment by Fizzadar** [comment]
I've added both of those in https://github.com/Fizzadar/pyinfra/commit/637629843dee015b48a5b905834671636df89140, due to be released in `v1.2` in the coming days!
